ui/src/overview.tsx

```diff
@@ -117,9 +120,54 @@ function AddCriteriaButton(props: { group: string | GroupKind }) {
   );
 }
 
-function ParticipantsGroup(props: { group: Group }) {
+function ParticipantsGroup(props: { group: Group; index: number }) {
+  const dispatch = useAppDispatch();
+  const cohort = useCohort();
+  const [renameGroupDialog, showRenameGroup] = useTextInputDialog({
+    title: "Edit Group Name",
+    initialText:
+      props.group.name === ""
+        ? "Group " + String(props.index + 1)
+        : props.group.name,
+    textLabel: "Group Name",
+    buttonLabel: "Confirm",
+    onConfirm: (name: string) => {
+      dispatch(
+        renameGroup({
+          cohortId: cohort.id,
+          groupId: props.group.id,
+          groupName: name,
+        })
+      );
+    },
+  });
+
+  const [groupMenu, groupShow] = useMenu({
+    children: [
+      <MenuItem key="1" onClick={showRenameGroup}>
+        Edit Group Name
+      </MenuItem>,
+    ],
+  });
+
   return (
     <Paper className="participants-group">
+      <Grid container className="group-title">
+        <Grid item xs="auto">
+          <IconButton onClick={groupShow} component="span" size="small">
+            <MoreVertIcon fontSize="small" />
+          </IconButton>
+          {groupMenu}
+          {renameGroupDialog}
+        </Grid>
+        <Grid item>
+          <Typography variant="h5" fontWeight="bolder">
```

Comment:
Remove `fontWeight`, even though it may look better. We want to keep formatting out of the code as much as possible so it's easier to customize it for different use cases.